modules/registry/registry-core/src/main/java/org/apache/airavata/registry/core/repositories/expcatalog/JobStatusRepository.java

```diff
@@ -113,4 +117,51 @@ public List<JobStatus> getDistinctListofJobStatus(String gatewayId,String status
         return  jobStatusRepository.selectWithNativeQuery(QueryConstants.FIND_JOB_COUNT_NATIVE_QUERY,
                 gatewayId,status,String.valueOf(time));
     }
+
+    public List<CpuUsage> getCpuUsages(String gatewayId, long fromTime, long toTime) {
+        JobStatusRepository jobStatusRepository = new JobStatusRepository();
+        List<CpuUsage> cpuUsages = new ArrayList<>();
+        List<JobStatus> startedJobStatusList = jobStatusRepository.selectWithNativeQuery(
+                QueryConstants.FIND_JOB_STATUS_STARTED_BEFORE_TIME, gatewayId, JobState.ACTIVE.name(),
+                String.valueOf(toTime));
+        ;
+        List<JobStatus> finishedJobStatusList = jobStatusRepository.selectWithNativeQuery(
+                QueryConstants.FIND_JOB_STATUS_FINISHED_AFTER_TIME, gatewayId, JobState.COMPLETE.name(),
+                JobState.FAILED.name(), JobState.CANCELED.name(), String.valueOf(fromTime));
+        Mapper mapper = ObjectMapperSingleton.getInstance();
+        for (JobStatus startedJobStatus : startedJobStatusList) {
+            for (JobStatus finishedJobStatus : finishedJobStatusList) {
+                JobStatusEntity startedJobStatusEntity = mapper.map(startedJobStatus, JobStatusEntity.class);
+                JobStatusEntity finishedJobStatusEntity = mapper.map(finishedJobStatus, JobStatusEntity.class);
+                if (startedJobStatusEntity.getJobId() == finishedJobStatusEntity.getJobId()) {
+                    String jobId = startedJobStatusEntity.getJobId();
+                    List<ExperimentModel> experiments = (new ExperimentRepository())
+                            .selectWithNativeQuery(QueryConstants.FIND_EXPERIMENT_WITH_JOB_ID, jobId);
+                    if (!experiments.isEmpty()) {
+                        ExperimentModel experiment = experiments.get(0);
+                        List<Integer> res = (new ExperimentRepository().selectWithNativeQuery(
```

Comment:
Do you need this query? You've already loaded the ExperimentModel so I think you can just use `experiment.getUserConfigurationData().getComputationalResourceScheduling().getTotalCPUCount()`.